platform/features/table/src/TableConfiguration.js

```diff
@@ -175,10 +185,15 @@ define(
              */
             this.getHeaders().forEach(function (columnTitle) {
                 configuration[columnTitle] =
-                    typeof defaultConfig[columnTitle] === 'undefined' ? true :
+                    typeof (defaultConfig || {})[columnTitle] === 'undefined' ? true :
```

Comment:
On line 179 you're already protecting against undefined default config; this change is redundant.

There is a lot of protection for undefined return values (breach of contract); this class should trust its private methods to follow their contract and not return undefined.  Why is it even possible that defaultColumnConfiguration can return a falsy value?